aws/ec2_key_pair.go

```diff
@@ -0,0 +1,67 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/hashicorp/go-multierror"
+)
+
+// getAllEc2KeyPairs extracts the list of existing ec2 key pairs.
+func getAllEc2KeyPairs(session *session.Session) ([]*string, error) {
```

Comment:
Ah that's nice. I wonder if that should be a separate ticket to handle? https://gruntwork.atlassian.net/browse/CORE-284 -- the acceptance criteria does not seem to mention about the capability to support such feature.

Reply:
I don't have a problem with adding it to a separate ticket, but I don't think we should release it without this functionality since all of the other resources support it.  Especially with a tool designed for destruction like cloud-nuke. It would be very easy for someone to create a config file with a regex and accidentally delete something because the regex feature wasn't added.

Reply:
Ah that's a great point. Let's add that as part of this if that's the case. Let me update the PR 👍